src/cli/commands/__tests__/lint.test.ts

```diff
@@ -1,448 +1,172 @@
-        .command(['lint', validCustomOas3SpecPath, '-r', 'http://foo.local/ruleset.yaml'])
-        .it('given valid remote ruleset file, outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
-    });
+  it('calls lint with document and default options', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
+        encoding: 'utf8',
+        format: 'stylish',
+      }),
+      undefined,
+    );
+  });
 
-    describe('when a standard oas3 ruleset provided through option', () => {
-      test
-        .stdout()
-        .command(['lint', invalidOas3SpecPath, '-r', standardOas3RulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain(
-            '1:1  warning  api-servers       OpenAPI `servers` must be present and non-empty array',
-          );
-          expect(ctx.stdout).toContain('3:6  warning  info-contact      Info object should contain `contact` object');
-          expect(ctx.stdout).toContain(
-            '3:6  warning  info-description  OpenAPI object info `description` must be present and non-empty string',
-          );
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('calls lint with document and custom encoding', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint --encoding utf16 ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
+        encoding: 'utf16',
+        format: 'stylish',
+      }),
+      undefined,
+    );
+  });
 
-      test
-        .stdout()
-        .command(['lint', oas2PetstoreSpecPath, '-r', standardOas2RulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain(
-            '46:24  warning  operation-description   Operation `description` must be present and non-empty string',
-          );
-          expect(ctx.stdout).toContain('22 problems (0 errors, 22 warnings, 0 infos)');
-          expect(ctx.stdout).toContain('OpenAPI 2.0 (Swagger) detected');
-        });
-    });
+  it('calls lint with document and custom encoding and format', async () => {
+    const doc = './__fixtures__/empty-oas2-document.json';
+    await run(`lint -f json --encoding utf16 ${doc}`);
+    expect(lint).toBeCalledWith(
+      doc,
+      expect.objectContaining({
+        encoding: 'utf16',
+        format: 'json',
+      }),
+      undefined,
+    );
   });
 
-  describe('when --quiet flag is provided', () => {
-    test
-      .stdout()
-      .command(['lint', invalidOas3SpecPath, '--quiet'])
-      .it('does not log any additional feedback', ctx => {
-        expect(ctx.stdout).not.toContain('OpenAPI 3.x detected');
-      });
+  it('calls lint with document and custom rulesets', async () => {
```

Comment:
```suggestion
  it('calls lint with document and custom ruleset', async () => {
```
later we have `calls lint with document and multiple custom rulesets`